operator/redisfailover/handler.go

```diff
@@ -81,6 +101,31 @@ func (r *RedisFailoverHandler) Handle(_ context.Context, obj runtime.Object) err
 		return err
 	}
 
+	// if last condition `Type` is not `Ready`, add it to the list
+	if rf.Status.Conditions[len(rf.Status.Conditions)-1].Type != redisfailoverv1.AppStateReady {
+
+		// make sure that all required resources are running as expected
+		if r.rfChecker.IsClusterRunning(rf) {
+
+			clusterCondition := redisfailoverv1.ClusterCondition{
+				Status:  redisfailoverv1.ConditionTrue,
+				Type:    redisfailoverv1.AppStateReady,
+				Message: "RedisFailover initialized successfully",
+			}
+
+			rf.Status.ObservedGeneration = rf.GetObjectMeta().GetGeneration()
+			rf.Status.AddCondition(clusterCondition)
+
+			_, err := r.rfService.UpdateStatus(rf)
+
+			if err != nil {
+				r.mClient.SetClusterError(rf.Namespace, rf.Name)
+				r.logger.Errorf("Can not update RedisFailover Status")
```

Comment:
Do we want to log this err like you did on line 76?

```suggestion
				r.logger.Errorf("Error attempting to update RedisFailover Status: %s", err)
```